Microsoft.Toolkit.Uwp.Input.GazeControls/GazeKeyboard.xaml.cs

```diff
@@ -0,0 +1,509 @@
+    /// </summary>
+    [Bindable]
+    public sealed partial class GazeKeyboard : UserControl
+    {
+        private InputInjector _injector;
+        private List<ButtonBase> _keyboardButtons;
+        private KeyboardPage _rootPage;
+        private TextPredictionGenerator _textPredictionGenerator;
+        private WordsSegmenter _wordsSegmenter;
+        private string _predictionLanguage;
+        private Button[] _predictionTargets;
+
+        /// <summary>
+        /// Gets or sets a value indicating whether the keys activate with gaze and dwell, or gaze and click
+        /// </summary>
+        public bool GazePlusClickMode { get; set; }
+
+        /// <summary>
+        /// Gets or sets the target text box for injecting keys
+        /// </summary>
+        public TextBox Target { get; set; }
+
+        /// <summary>
+        /// Gets or sets the prediction language
+        /// </summary>
+        public string PredictionLanguage
+        {
+            get
+            {
+                return _predictionLanguage;
+            }
+
+            set
+            {
+                _predictionLanguage = value;
+                _textPredictionGenerator = new TextPredictionGenerator(value);
+                _textPredictionGenerator.InputScope = Windows.UI.Text.Core.CoreTextInputScope.Text;
+                _wordsSegmenter = new WordsSegmenter(value);
+            }
+        }
+
+        private Uri _layoutUri;
+
+        /// <summary>
+        /// Gets or sets the URI of the layout file for the keyboard
+        /// </summary>
+        public Uri LayoutUri
+        {
+            get
+            {
+                return _layoutUri;
+            }
+
+            set
+            {
+                _layoutUri = value;
+                _ = LoadLayout(value);
+            }
+        }
+
+        /// <summary>
+        /// Gets or sets the prediction targets
+        /// </summary>
+        public Button[] PredictionTargets
+        {
+            get
+            {
+                return _predictionTargets;
+            }
+
+            set
+            {
+                if (_predictionTargets != null)
+                {
+                    foreach (var target in _predictionTargets)
+                    {
+                        target.Click -= OnPredictionSelected;
+                    }
+                }
+
+                _predictionTargets = value;
+                foreach (var target in _predictionTargets)
+                {
+                    target.Click += OnPredictionSelected;
+                }
+            }
+        }
+
+        /// <summary>
+        /// Initializes a new instance of the <see cref="GazeKeyboard"/> class.
+        /// </summary>
+        public GazeKeyboard()
+        {
+            InitializeComponent();
+            PredictionLanguage = "en-US";
+            _injector = InputInjector.TryCreate();
+        }
+
+        internal static void FindChildren<T>(List<T> results, DependencyObject startNode)
```

Comment:
Again, use the existing functions in the Toolkit, I believe @Sergio0694 added one which can iterate and run functions over the whole tree.

Reply:
Can you please share a pointer to what I should be using?

Reply:
Found it and replaced it. Thanks.